File: Java/Working with Numbers/37. Octal to Binary conversion.java
import java.util.Scanner;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter the Octal value:");
        String octal = userInput.nextLine();
        int decimal = Integer.parseInt(octal, 8);
        String binary = Integer.toBinaryString(decimal);
        
        // Method 2
        // int octal = userInput.nextInt();
		// int decimal = 0;
		// int n = 0;
        // if (octal == 0) {
        //     System.out.println("The Binaey value of 0 is 0");
        // }
		// while(octal > 0) {
		// 	int digit = octal % 10;
		// 	decimal += digit * Math.pow(8, n);
		// 	octal /= 10;
		// 	n += 1;
		// }
        
        // String binary = "";
        // while(decimal > 0) {
        //     int rem = decimal % 2;
        //     binary = rem + binary;
        //     decimal /= 2;
        // }        
        System.out.println("The Binary value of "+octal +" is " +binary);
        userInput.close();
    }
}

File: Python/02. Working with Numbers/37. Octal to Binary conversion.py
octal = input("Enter the Octal Value :") # use int() for Method 2
binary = bin(int(octal, 8))[2:]

print("Binary value of", octal, "is", binary)

# Method 2

# i = 0
# decimal = 0
# while (octal != 0): # Octal to Decimal
#     digit = octal % 10
#     decimal += digit * pow(8, i)
#     octal //= 10
#     i += 1

# binary = 0
# rem = 0
# i = 1
# while (decimal != 0): # Decimal to Binary
#     rem = decimal % 2
#     binary += rem * i
#     decimal //= 2
#     i *= 10
# print("Binary value is", binary)